frontend/catalyst/jax_tracer.py

```diff
@@ -870,7 +891,21 @@ def trace_quantum_measurements(
                     result = sample_p.bind(obs_tracers, shots=shots, shape=shape)
                     if using_compbasis:
                         result = jnp.astype(result, jnp.int64)
-                    out_classical_tracers.append(result)
+
+                    reshaped_result = ()
+                    shot_vector = get_device_shot_vector(device)
+                    start_idx = 0  # Start index for slicing
+                    for shot, copies in shot_vector:
+                        for _ in range(copies):
+                            sliced_result = result[start_idx : start_idx + shot]
+                            reshaped_result += (sliced_result.reshape(shot, nqubits),)
+                            start_idx += shot
+
+                    if len(reshaped_result) == 1:
+                        reshaped_result = reshaped_result[0]
+
+                    out_classical_tracers.append(reshaped_result)
```

Comment:
The reshape looks good, I let @dime10 double check :+1:

Reply:
Looks great to me 🎉